ktor-server/ktor-server-test-host/nix/src/EngineTestBaseNix.kt

```diff
@@ -29,7 +29,7 @@ actual abstract class EngineTestBase<TEngine : ApplicationEngine, TConfiguration
     actual val applicationEngineFactory: ApplicationEngineFactory<TEngine, TConfiguration>
 ) : CoroutineScope {
     private val testJob = Job()
-    actual override val coroutineContext: CoroutineContext = testJob + Dispatchers.Main
+    actual override val coroutineContext: CoroutineContext = testJob + Dispatchers.Unconfined
```

Comment:
Please consider making `expect`-`actual` method for dispatcher. I would also avoid using `Unconfined` here

Reply:
Main is available only on `darwin` targets on `nix`. May be we can use `Default` on all `nix` targets?
I was used `Unconfined` here instead of `Default`, as `ktor-network` sockets closing somehow failing the whole process, but I have not found the root cause of it.
I will try to investigate more

Reply:
Didn't notice `Main` here. We should replace it with `IO` or `Default`

Reply:
FYI, it's `nix` specific code.
There is no `IO` on native on current moment, so `Default` is the best choice, as it's also used as JVM replacement of `IO` on `nix` in `ktor`

Reply:
replaced with Default, but anyway I need to use `Unconfined` in one test (testWebSocketDisconnectDuringSending) for using ktor-network socket in it. If not to do it, then it will fail process with no exception thrown.
For now, I can't find the root cause of it :(